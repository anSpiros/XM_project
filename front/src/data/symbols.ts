import { SymbolsInterface } from "../interfaces/symbols.interface";

export const symbols: SymbolsInterface[] = [
  {
    value: "AAPL",
    label: "Apple Inc.",
  },
  {
    value: "MSFT",
    label: "AppleMicrosoft Corporation",
  },
  {
    value: "AMZN",
    label: "Amazon.com",
  },
  {
    value: "GOOGL",
    label: "Alphabet Inc.",
  },
  {
    value: "FB",
    label: "Facebook, Inc.",
  },
  {
    value: "TCEHY",
    label: "Tencent Holdings Ltd.",
  },
  {
    value: "TSM",
    label: "Taiwan Semiconductor",
  },
];
