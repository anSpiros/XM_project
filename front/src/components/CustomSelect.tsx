import React from "react";
import { SymbolsInterface } from "../interfaces/symbols.interface";

interface CustomSelectProps {
  symbols: SymbolsInterface[];
  formRef: any;
}

function CustomSelect({ symbols, formRef }: CustomSelectProps) {
  return (
    <>
      <select
        id="symbol"
        defaultValue=""
        name="symbol"
        ref={formRef}
        className="custom-select"
      >
        <option value="" disabled>
          Select
        </option>
        {symbols.map((data) => {
          return (
            <option key={data.value} value={data.value}>
              {data.label}
            </option>
          );
        })}
      </select>
    </>
  );
}

export default CustomSelect;
