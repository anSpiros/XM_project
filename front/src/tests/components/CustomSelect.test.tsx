import React from "react";
import { cleanup, render } from "@testing-library/react";
import CustomSelect from "../../components/CustomSelect";
import { symbols } from "../../data/symbols";

describe("Custom select component", () => {
  afterEach(cleanup);
  test("should render correctly", () => {
    const { asFragment } = render(
      <CustomSelect symbols={symbols} formRef={React.createRef()} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
