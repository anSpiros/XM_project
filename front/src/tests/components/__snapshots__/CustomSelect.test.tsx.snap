// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom select component should render correctly 1`] = `
<DocumentFragment>
  <select
    class="custom-select"
    id="symbol"
    name="symbol"
  >
    <option
      disabled=""
      selected=""
      value=""
    >
      Select
    </option>
    <option
      value="AAPL"
    >
      Apple Inc.
    </option>
    <option
      value="MSFT"
    >
      AppleMicrosoft Corporation
    </option>
    <option
      value="AMZN"
    >
      Amazon.com
    </option>
    <option
      value="GOOGL"
    >
      Alphabet Inc.
    </option>
    <option
      value="FB"
    >
      Facebook, Inc.
    </option>
    <option
      value="TCEHY"
    >
      Tencent Holdings Ltd.
    </option>
    <option
      value="TSM"
    >
      Taiwan Semiconductor
    </option>
  </select>
</DocumentFragment>
`;
